localstack/services/apigateway/helpers.py

```diff
@@ -1185,3 +1200,58 @@ def add_path(path, parts, parent_id=""):
         rest_api.endpoint_configuration = endpoint_config
 
     return rest_api
+
+
+def apply_template(
```

Comment:
Thinking on making this explicit for request templates similar to `apply_request_response_templates`. Also, have these template applications inside the integrations, part of the refactor we have discussed before.

wip: backend integrations refactor https://github.com/localstack/localstack-ext/compare/master...backend_integration_refactor

What you think @whummer?